### PR TITLE
feat: list protocols on DefiLlama

### PR DESCRIPTION
**Pull request by Addis-Belete**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/clearpool: 
        
 
  
    
    
    [object Object]

**Comment by Addis-Belete** [comment]
@georgeciubotaru this are the list of keys that are passed to chains. and I am getting error while trying to pass `protocol`(`module.exports[chain][protocol]`)
Screenshot 2024-05-10 at 7 41 44 PM

**Comment by georgeciubotaru** [comment]
@Addis-Belete lets then to sum the tvl for all protocol per network and return 1 tvl value

**Comment by Addis-Belete** [comment]
Log: getting Error: Llama RPC error! methos: undefined 
Screenshot 2024-05-13 at 10 20 15 AM

**Comment by Addis-Belete** [comment]
@georgeciubotaru review this PR and let us create a PR on the main repo and get help from team on the above error(https://github.com/clearpool-finance/DefiLlama-Adapters/pull/10#issuecomment-2106833570) from test case.

File: projects/clearpool/index.js [hunk 1]
```diff
@@ -11,44 +11,107 @@ const CPOOL = "0x66761fa41377003622aee3c7675fc7b5c1c2fac5";
...
-}
-
-Object.keys(config).forEach(chain => {
-  const { fromBlock, factory } = config[chain]
-
-  const tvl = async (api) => {
-    const { pools, tokens } = await _getLogs(api)
-    return api.sumTokens({ tokensAndOwners2: [tokens, pools] })
-  }
+  dynamic: {
```
Comment (georgeciubotaru) [inline projects/clearpool/index.js:29]: @Addis-Belete First can we use their prettier or `eslint` and format the file?

Comment (georgeciubotaru) [reply projects/clearpool/index.js:29]: Second, let's use the`Chain` constant as it's here:
https://github.com/DefiLlama/dimension-adapters/blob/master/protocols/uniswap/index.ts

**Comment by llamatester** [comment]
Error while running adapter at projects/clearpool: 
        
 
  
    
    
    [object Object]

**Comment by llamatester** [comment]
Error while running adapter at projects/clearpool: 
        
 
  
    
    
    [object Object]

File: projects/clearpool/index.js [hunk 2]
```diff
@@ -11,44 +11,120 @@ const CPOOL = "0x66761fa41377003622aee3c7675fc7b5c1c2fac5";
...
+const CHAIN = {
+  ARBITRUM: "arbitrum",
+  AVAX: "avax",
+  ETHEREUM: "ethereum",
+  OPTIMISM: "optimism",
+  POLYGON: "polygon",
+  POLYGON_ZKEVM: "polygon_zkevm",
+  FLARE: "flare",
+  BASE: "base"
+};
```
Comment (georgeciubotaru) [inline projects/clearpool/index.js]: is it possible to use directly the `sdk` package without declaring internally?

**Comment by llamatester** [comment]
The adapter at projects/clearpool exports TVL: 
        
 
 
    ethereum-staking          18.78 M
    staking                   18.78 M
    borrowed                  16.99 M
    optimism-borrowed         10.81 M
    ethereum-borrowed         5.40 M
    optimism                  2.03 M
    ethereum                  1.83 M
    avax-borrowed             610.64 k
    polygon_zkevm-borrowed    174.51 k
    polygon                   41.17 k
    polygon_zkevm             32.22 k
    avax                      45
    polygon-borrowed          0
    flare                     0
    flare-borrowed            0
    arbitrum-borrowed         0
    arbitrum                  0
    base                      0
    base-borrowed             0
    
    total                    3.93 M

File: projects/clearpool/index.js (hunk 2 above)
Comment (georgeciubotaru) [reply projects/clearpool/index.js]: @Addis-Belete How about `Mantle`?

**Comment by llamatester** [comment]
The adapter at projects/clearpool exports TVL: 
        
 
 
    ethereum-staking          18.80 M
    staking                   18.80 M
    borrowed                  16.99 M
    optimism-borrowed         10.81 M
    ethereum-borrowed         5.40 M
    optimism                  2.03 M
    ethereum                  1.83 M
    avax-borrowed             610.64 k
    polygon_zkevm-borrowed    174.50 k
    polygon                   41.17 k
    polygon_zkevm             32.22 k
    avax                      45
    polygon-borrowed          0
    arbitrum                  0
    arbitrum-borrowed         0
    flare-borrowed            0
    flare                     0
    base                      0
    base-borrowed             0
    
    total                    3.93 M

File: projects/clearpool/index.js (hunk 2 above)
Comment (Addis-Belete) [reply projects/clearpool/index.js]: @georgeciubotaru No, the networks are listed in the helper/chains as an array.

**Comment by llamatester** [comment]
The adapter at projects/clearpool exports TVL: 
        
 
 
    ethereum-staking          18.81 M
    staking                   18.81 M
    borrowed                  17.61 M
    optimism-borrowed         10.81 M
    ethereum-borrowed         5.70 M
    optimism                  2.03 M
    ethereum                  1.53 M
    avax-borrowed             610.64 k
    mantle-borrowed           315.65 k
    polygon_zkevm-borrowed    174.48 k
    polygon                   41.17 k
    mantle                    35.84 k
    polygon_zkevm             32.22 k
    avax                      45
    polygon-borrowed          0
    arbitrum                  0
    arbitrum-borrowed         0
    base                      0
    flare-borrowed            0
    flare                     0
    base-borrowed             0
    
    total                    3.67 M

**Comment by llamatester** [comment]
The adapter at projects/clearpool exports TVL: 
        
 
 
    ethereum-staking          18.80 M
    staking                   18.80 M
    borrowed                  17.61 M
    optimism-borrowed         10.81 M
    ethereum-borrowed         5.70 M
    optimism                  2.03 M
    ethereum                  1.53 M
    avax-borrowed             610.51 k
    mantle-borrowed           315.65 k
    polygon_zkevm-borrowed    174.41 k
    polygon                   41.17 k
    mantle                    35.84 k
    polygon_zkevm             32.20 k
    avax                      45
    polygon-borrowed          0
    arbitrum-borrowed         0
    arbitrum                  0
    flare                     0
    flare-borrowed            0
    base                      0
    base-borrowed             0
    
    total                    3.67 M

**Comment by georgeciubotaru** [comment]
@Addis-Belete 1 thing to check. 

We need to have such functionality:

ethereum(tvl) = dynamicTvl + vaultsTvl + treasuryTvl
ethereum(Borrowed) = dynamicBorrow + vaultsBorrow + treasuryBorrows

**Comment by Addis-Belete** [comment]
@georgeciubotaru Can we merge this, and resolve the comment in separate PR?

**Review by georgeciubotaru — APPROVED** [review]

**Pull request merged**